test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/BasicSearchTests.cs

```diff
@@ -192,6 +164,68 @@ public async Task GivenVariousTypesOfResources_WhenSearchingAcrossAllResourceTyp
         }
 
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
+        public async Task GivenMultiplePagesOfVariousTypesOfResourcesInSql_WhenUsingTypeParameterToSearchForMultipleResourceTypes_ThenCorrectResourcesAreReturned()
+        {
+            // Create various resources.
+            string tag = Guid.NewGuid().ToString();
+            Patient[] patients = await Client.CreateResourcesAsync<Patient>(
+                p => SetPatientInfo(p, "city1", "name1", tag),
+                p => SetPatientInfo(p, "city2", "name2", tag),
+                p => SetPatientInfo(p, "city3", "name3", tag),
+                p => SetPatientInfo(p, "city4", "name4", tag));
+
+            Observation[] observations = await Client.CreateResourcesAsync<Observation>(
+                o => SetObservationInfo(o, tag),
+                o => SetObservationInfo(o, tag));
+
+            List<Resource> expectedResources = new List<Resource>();
+            foreach (Observation observation in observations)
+            {
+                expectedResources.Add(observation);
+            }
+
+            foreach (Patient patient in patients)
+            {
+                expectedResources.Add(patient);
+            }
+
+            await ExecuteAndValidateBundle($"?_type=Patient,Observation&_tag={tag}&_count=3", sort: false, pageSize: 3, expectedResources.ToArray());
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.CosmosDb)]
+        public async Task GivenMultiplePagesOfVariousTypesOfResourcesInCosmos_WhenUsingTypeParameterToSearchForMultipleResourceTypes_ThenCorrectResourcesAreReturned()
+        {
+            // Create various resources.
+            string tag = Guid.NewGuid().ToString();
+            Patient[] patients = await Client.CreateResourcesAsync<Patient>(
+                p => SetPatientInfo(p, "city1", "name1", tag),
+                p => SetPatientInfo(p, "city2", "name2", tag),
+                p => SetPatientInfo(p, "city3", "name3", tag),
+                p => SetPatientInfo(p, "city4", "name4", tag));
+
+            Observation[] observations = await Client.CreateResourcesAsync<Observation>(
+                o => SetObservationInfo(o, tag),
+                o => SetObservationInfo(o, tag));
+
+            List<Resource> expectedResources = new List<Resource>();
+            foreach (Patient patient in patients)
+            {
+                expectedResources.Add(patient);
+            }
+
+            foreach (Observation observation in observations)
+            {
+                expectedResources.Add(observation);
+            }
+
+            await ExecuteAndValidateBundle($"?_type=Patient,Observation&_tag={tag}&_count=3", sort: false, pageSize: 3, expectedResources.ToArray());
```

Comment:
>ExecuteAndValidateBundle [](http://example.com/codeflow?start=18&length=24)

nit: can we sort by lastupdated and have one test?

Reply:
I had two separate tests because I wanted to highlight the difference in behavior between Cosmos and SQL for such queries.